### PR TITLE
Fix Histograms and reload volumes

### PR DESCRIPTION
**Pull request by kmilo9999**
- fix histograms for both volume representations. 
- fix ui - clear all button and reaload volume funcitonalities

### CODE REVIEW

File: include/render/Volume.h [hunk 1]
```diff
@@ -234,7 +234,7 @@ class Volume
 
 	bool m_texture_initialized;
 	unsigned int m_texture_id;
-	unsigned int m_texture_2_id;
+
```
Comment (RobertGemmaJr) [inline include/render/Volume.h:237]:
```suggestion
```

File: include/vrapp/VolumeVisualizationApp.h [hunk 2]
```diff
@@ -80,7 +80,7 @@ class VolumeVisualizationApp : public MinVR::VRApp
 
   std::chrono::steady_clock::time_point m_lastTime;
 
-  VRVolumeApp *m_vrVolumeApp;
+  std::unique_ptr<VRVolumeApp> m_vrVolumeApp;
```
Comment (RobertGemmaJr) [inline include/vrapp/VolumeVisualizationApp.h:83]: What's this `unique_ptr` do? Does it prevent you from having multiple pointers to the same object? I've never seen it before

File: src/interaction/LabelsManager.cpp [hunk 3]
```diff
@@ -98,15 +99,15 @@ void LabelsManager::add(std::string &texture_path, float x, float y, float z, fl
   if (m_texture_cache.find(texture_path) == m_texture_cache.end())
   {
 
-    Texture *texture = new Texture(GL_TEXTURE_2D, texture_path);
+      std::shared_ptr<Texture> texture = std::make_shared< Texture>(GL_TEXTURE_2D, texture_path);
```
Comment (RobertGemmaJr) [inline src/interaction/LabelsManager.cpp:102]: And this means their can be multiple copies of the same pointer in the app?

File: src/vrapp/VRVolumeApp.cpp [hunk 4]
```diff
@@ -569,27 +569,29 @@ void VRVolumeApp::render(const MinVR::VRGraphicsState& render_state)
...
-		 glm::mat4 mesh_model_matrix = glm::translate(volume_mv, glm::vec3(-0.5f, -0.5f, -0.5f * px / pz));
-		 mesh_model_matrix = glm::scale(mesh_model_matrix, glm::vec3(px, py, px));
-		**/
-
-		//volume_mv = glm::rotate(volume_mv, glm::radians(180.0f), glm::vec3(1.0f,0.f,0.f));
-		volume_mv = glm::translate(volume_mv, glm::vec3(-0.5f, -0.5f, -0.5f * px / pz));
-		volume_mv = glm::scale(volume_mv, glm::vec3(px, py, px));
-
-		mesh->get_model().setMVMatrix(volume_mv);
+		if (!m_volumes.empty())
```
Comment (RobertGemmaJr) [inline src/vrapp/VRVolumeApp.cpp:572]: Could you do an `if(m_volumes.empty()) continue`? Or do you prefer this syntax?

File: src/vrapp/VRVolumeApp.cpp [hunk 5]
```diff
@@ -901,6 +903,7 @@ void VRVolumeApp::clear_data()
 		}
 	}
 	m_volumes.clear();
+	m_numVolumes = 0;
```
Comment (RobertGemmaJr) [inline src/vrapp/VRVolumeApp.cpp:906]: I think this may be an used variable?

**Review by RobertGemmaJr — APPROVED** [review]
Looks great! Just a couple questions for my own learning

File: include/vrapp/VolumeVisualizationApp.h (hunk 2 above)
Comment (kmilo9999) [reply include/vrapp/VolumeVisualizationApp.h:83]: [That's something fancy from C++ 11](https://learn.microsoft.com/en-us/cpp/cpp/smart-pointers-modern-cpp?view=msvc-170) to delegate memory allocation to the compiler and free the user from that responsibility.  Im trying to make this old code a little bit more modern.

File: src/interaction/LabelsManager.cpp (hunk 3 above)
Comment (kmilo9999) [reply src/interaction/LabelsManager.cpp:102]: Yup. All is managed by the compiler.

File: src/vrapp/VRVolumeApp.cpp (hunk 4 above)
Comment (kmilo9999) [reply src/vrapp/VRVolumeApp.cpp:572]: Let me keep it this way for now. I need this piece of code to solve #60

File: src/vrapp/VRVolumeApp.cpp (hunk 5 above)
Comment (kmilo9999) [reply src/vrapp/VRVolumeApp.cpp:906]: Not really. That variable is needed by other methods. `void VRVolumeApp::init_num_volumes(int nVolumes)`.

**Pull request merged**